ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/DebugTraceBlockByNumber.java

```diff
@@ -75,22 +98,68 @@ protected Object resultByBlockNumber(
           RpcErrorType.INVALID_TRANSACTION_TRACE_PARAMS,
           e);
     }
+    Optional<Block> maybeBlock =
+        getBlockchainQueries().getBlockchain().getBlockByNumber(blockNumber);
 
-    return blockHash
+    return maybeBlock
         .flatMap(
-            hash ->
+            block ->
                 Tracer.processTracing(
```

Comment:
Why can't this reuse the same call made in AbstractDebugTraceBlock (as ByHash appears to)?

Reply:
Good question, this is because DebugTraceBlockByNumber already inherit from AbstractBlockParameterMethod, to handle flags related to block number, like latest, safe or finalized.